src/array.c

```diff
@@ -457,13 +457,39 @@ JL_DLLEXPORT jl_array_t *jl_pchar_to_array(const char *str, size_t len)
     return a;
 }
 
+void jl_set_in_flight_bit_for_array_to_string(jl_array_t *a)
+{
+    uintptr_t msk = (1 << ARRAY_TO_STRING_IN_FLIGHT_BIT_OFFSET);
+    uintptr_t header = jl_atomic_fetch_or((_Atomic(uintptr_t) *)jl_astaggedvalue(a), msk);
+    if (header & msk) {
+        // Race detected... Someone already set the in-flight bit.
+        // TODO(netto): figure out how to pretty-print this into JSON in order
+        // for this to show up properly on DataDog.
+        jl_safe_printf("ERROR: racy conversion of array to string... Setting in-flight bit\n");
```

Comment:
Cool! 

How will this help detect the source of the race though? We need to know the (julia) call-site, right?

Reply:
See the TODO. I will augment this to print the backtrace.